hail/python/hail/matrixtable.py

```diff
@@ -3570,8 +3570,8 @@ def union_rows(*datasets: 'MatrixTable', _check_cols=True) -> 'MatrixTable':
                                      f"Datasets 0 and {wrong_keys+1} have different columns (or possibly different order).")
             return MatrixTable(MatrixUnionRows(*[d._mir for d in datasets]))
 
-    @typecheck_method(other=matrix_table_type)
-    def union_cols(self, other: 'MatrixTable') -> 'MatrixTable':
+    @typecheck_method(other=matrix_table_type, _outer=bool)
+    def union_cols(self, other: 'MatrixTable', _outer=False) -> 'MatrixTable':
```

Comment:
why not make this public?

Reply:
Just need a better test and some docs and we can make it public.

Reply:
Sure. I was just being cautious until we'd had more discussion about the interface. Do you like `outer`, or should it be more explicit like `outer_join_rows`?

Reply:
I like `row_join_type` or something that takes a string, with a default of inner.

That way we can add left/right if people want that (reasonable)

Reply:
Seems good to me